lib/category_list_serializer_extension.rb

```diff
@@ -4,6 +4,18 @@ module GlobalFilter::CategoryListSerializerExtension
 
   def categories
     filtered_categories = GlobalFilter::FilterTag.categories_for_tag(filter_tag, scope)
-    filtered_categories.any? ? filtered_categories : object.categories
+    category_stats = GlobalFilter::FilterTag.find_by(name: filter_tag)&.category_stats
+    categories = filtered_categories.any? ? filtered_categories : object.categories
+
+    return categories if category_stats.nil?
+    update_topic_and_post_count(category_stats)
+    categories
+  end
+
+  def update_topic_and_post_count(category_stats)
```

Comment:
@pmusaraj do you have any ideas that come to mind for performance improvements here? This is providing correct topic/post counts per category, but it could be faster for sure...